scripts/populate-popular-and-new-and-noteworthy-projects.py

```diff
@@ -0,0 +1,116 @@
+"""
+This will update node links on POPULAR_LINKS_NODE and NEW_AND_NOTEWORTHY_LINKS_NODE.
+"""
+import sys
+import json
+import urllib2
+import logging
+import datetime
+import dateutil
+import operator
+from modularodm import Q
+from website.app import init_app
+from website import models
+from framework.auth.core import Auth
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+from website.settings.defaults import POPULAR_LINKS_NODE, NEW_AND_NOTEWORTHY_LINKS_NODE
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_data(url):
+    """ Fetch data and decode json """
+    response = urllib2.urlopen(url)
+    data = json.load(response)
+    return data
+
+
+def get_popular_nodes():
+    """ Fetch data from url that returns dict with a list of popular nodes from piwik """
+    # TODO Shouldn't hardcode URL - only works locally
+    discover_url = 'http://127.0.0.1:5000/api/v1/explore/activity/popular/raw/'
+    return retrieve_data(discover_url)
+
+
+def get_new_and_noteworthy_nodes():
+    """ Fetches nodes created in the last month and returns 25 sorted by highest log activity """
+    # TODO Shouldn't hardcode URL - only works locally
+    today = datetime.datetime.now()
+    last_month = (today - dateutil.relativedelta.relativedelta(months=1)).isoformat()
+    discover_url = 'http://127.0.0.1:8000/v2/nodes/?sort=-date_created&page[size]=1000&related_counts=True&filter[date_created][gt]={}'.format(last_month)
+    data = retrieve_data(discover_url)['data']
+    node_log_count_mapping = {}
+    for new_node in data:
+        node_log_count_mapping[new_node['id']] = new_node['relationships']['logs']['links']['related']['meta']['count']
+    sort_by_log_count = sorted(node_log_count_mapping.items(), key=operator.itemgetter(1), reverse=True)
+    sorted_node_ids = [node[0] for node in sort_by_log_count]
+    return sorted_node_ids[:25]
+
+def is_eligible_node(node):
+    """
+    Check to ensure that node is not the POPULAR or NEW_AND_NOTEWORTHY LINKS_NODE.
+    Ensures QA members are not contributors
+    """
+    if node._id == POPULAR_LINKS_NODE or node._id == NEW_AND_NOTEWORTHY_LINKS_NODE:
+        return False
+
+    qa_user_ids = ['nxygz', 'x952z', 'tgak8', 'gaexu', 'rgc49', 'nsx26', 'j52af', 'rbk3c', 'xyubm', 'bje5z', 'twkqb', 'mvzr6', 'dihba']
```

Comment:
Do you have any ideas? We don't want to just filter out all accounts with a @cos.io email account, for example, because there are many people in the office working on legitimate projects on the OSF, and this could filter out projects like this: https://osf.io/ezcuj/.